debugstatus/handler.go

```diff
@@ -41,6 +42,15 @@ type Handler struct {
 	// of the endpoints under /debug/pprof - the
 	// error returned will be ErrNoPprofConfigured.
 	CheckPprofAllowed func(req *http.Request) error
+
+	// CheckTraceAllowed will be used to check whether the given
+	// trace request should be allowed. It should return an error if
+	// not, which will not be masked. If this is nil, no access will
+	// be allowed to either /debug/events or /debug/requests - the
+	// error returned will be ErrNoTraceConfigured. If access is
+	// allowed the sensative boolean is passed on the the trace
```

Comment:
Perhaps:

// If access is allowed, the sensitive value specifies whether
// sensitive trace events will be shown.

?